Cython/Includes/cpython/datetime.pxd

```diff
@@ -161,6 +172,27 @@ cdef inline object datetime_new(int year, int month, int day, int hour, int minu
 cdef inline object timedelta_new(int days, int seconds, int useconds):
     return PyDateTimeAPI.Delta_FromDelta(days, seconds, useconds, 1, PyDateTimeAPI.DeltaType)
 
+# Create a timezone from an offset, optionally a name.
+cdef inline object timezone_new(offset, name):
+    return PyDateTimeAPI.TimeZone_FromTimeZone(offset, name)
+
+# Create a datetime object with a fold using DateTime CAPI factory function.
+cdef inline object datetime_and_fold_new(int year, int month, int day, int hour, int minute, int second, int microsecond, object tz, int fold):
+    return PyDateTimeAPI.DateTime_FromDateAndTimeAndFold(year, month, day, hour, minute, second, microsecond, tz, fold, PyDateTimeAPI.DateTimeType)
```

Comment:
`fold` defaults to `0`. If it wasn't added only in Py3.6 and later, we could simply make it an optional argument to the existing `datetime_new()` function. However, the fact that the `…AndFold()` C-API functions are not available at all in Py<3.6 requires a bit of C macro magic to make this compile in older Pythons.

Reply:
Are you proposing that something similar to what was done earlier in the file is done? I am referencing the following bit of code (in the doc-strings):

```
    """
    #if PY_MAJOR_VERSION < 3 && !defined(PyDateTime_DELTA_GET_DAYS)
    #define PyDateTime_DELTA_GET_DAYS(o)         (((PyDateTime_Delta*)o)->days)
    #endif
    #if PY_MAJOR_VERSION < 3 && !defined(PyDateTime_DELTA_GET_SECONDS)
    #define PyDateTime_DELTA_GET_SECONDS(o)      (((PyDateTime_Delta*)o)->seconds)
    #endif
    #if PY_MAJOR_VERSION < 3 && !defined(PyDateTime_DELTA_GET_MICROSECONDS)
    #define PyDateTime_DELTA_GET_MICROSECONDS(o)   (((PyDateTime_Delta*)o)->microseconds)
    #endif
    """
```

This should probably be done for the upcoming `get_utc()` function then as well?

Reply:
Yes. We'd define them as an alias to the non-`AndFold` functions, for anything before 3.6(beta? – or whenever it was added).

Not sure if we should make it an error if a non-0 value is passed, but I'd rather not. It seems reasonable for user code to ignore this case, without special treatment. Making it an error, while making it visible that the case is not handled correctly, would also require the users who do not care to add special-casing code on their side.